[ie/NascarClassics] Add extractor

JrM2628 opened the pull request:
### Description of your *pull request* and other information

Adds support for [NASCAR Classics](https://classics.nascar.com) videos. 

Note: I named the file `nascar.py` instead of `nascarclassics.py` in the event that someone creates a `nascar.com` information extractor in the future.  

<details open><summary>Template</summary> <!-- OPEN is intentional -->

<!--
    # PLEASE FOLLOW THE GUIDE BELOW

    - You will be asked some questions, please read them **carefully** and answer honestly
    - Put an `x` into all the boxes `[ ]` relevant to your *pull request* (like [x])
    - Use *Preview* tab to see what your *pull request* will actually look like
-->

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check those that apply and remove the others:
- [x] I am the original author of the code in this PR, and I am willing to release it under [Unlicense](http://unlicense.org/)
- [x] I have read the [policy against AI/LLM contributions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#automated-contributions-ai--llm-policy) and understand I may be blocked from the repository if it is violated

### What is the purpose of your *pull request*? Check those that apply and remove the others:
- [x] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))

</details>

seproDev commented on yt_dlp/extractor/nascar.py:
```diff
@@ -0,0 +1,63 @@
+            'tags': tags,
+            'timestamp': timestamp,
+            'chapters': chapters,
+        }
```
Can be simplified to a single `traverse_obj` call. m3u8 url extraction might as well be fatal
```suggestion
        return {
            'id': video_id,
            'formats': self._extract_m3u8_formats(content_data['input']['src'], video_id),
            **traverse_obj(content_data, {
                'title': ('input', 'name', {str}),
                'description': ('input', 'description', {str}),
                'thumbnail': ('input', 'thumbnail', {url_or_none}),
                'tags': ('input', 'settings', 'tags', ..., {str}),
                'timestamp': ('input', 'start_time', {parse_iso8601}),
                'chapters': ('overlay', 'data', 'timelines', 0, 'events', ..., {
                    'start_time': ('timestamp', {float_or_none}),
                    'title': ('name', {str}),
                }),
            }),
        }
```

seproDev commented on yt_dlp/extractor/nascar.py:
```diff
@@ -0,0 +1,63 @@
+            'chapters': 'count:18',
+        },
+    },
+        {
```
```suggestion
    }, {
```

JrM2628 commented:
Thanks for the suggestions, @seproDev. Implemented and modified imports. Are we okay with the nested key access in line 41? 

```
content_data = self._search_nextjs_data(
            webpage, video_id)['props']['pageProps']['contentData']
```

bashonly commented on yt_dlp/extractor/nascar.py:
```diff
@@ -0,0 +1,57 @@
+
+        return {
+            'id': video_id,
+            'formats': self._extract_m3u8_formats(content_data['input']['src'], video_id),
```
I would do this just in case, unless you know the videos can have a different extension. It can prevent videos downloading with an m3u8 extension sometimes
```suggestion
            'formats': self._extract_m3u8_formats(content_data['input']['src'], video_id, 'mp4'),
```
